app/views/enrollments/new.html.erb

```diff
@@ -0,0 +1,9 @@
+<div class="container">
+  <h1>New enrollment for <%= @course.title %></h1>
+
+  <%= simple_form_for @enrollment do |f| %>
```

Comment:
, url: course_enrollments_path(@course) 

without this, the form send you to /enrollment (normal POST when the enrollment is not nested inside another route).